sympy/core/logic.py

```diff
@@ -11,6 +11,32 @@
 from sympy.core.compatibility import range
 
 
+def _torf(args):
+    """Return True if all args are True, False if they
+    are all False, else None.
+
+    >>> from sympy.core.logic import _torf
+    >>> _torf((True, True))
+    True
+    >>> _torf((False, False))
+    False
+    >>> _torf((True, False))
+    """
+    sawT = sawF = False
```

Comment:
The following is probably as efficient (or more so):

```
if all(a is True for a in args): return True
if all(a is False for a in args): return False
```

At worst, all are True and the last one is False so the 2nd `if` runs for 1 arg. At minimum, for a case for args starting with 2 None-s, each `if` will run once.